.github/workflows/ci_windows_x64_msvc.yml

```diff
@@ -44,9 +47,37 @@ jobs:
           python3 -m pip install -r runtime/bindings/python/iree/runtime/build_requirements.txt
       - name: "Installing requirements"
         run: choco install ccache --yes
+      - name: "Installing MSVC requirements"
+        run: |
+          choco install visualstudio2022buildtools -y
+          choco install visualstudio2022community --package-parameters "--add Microsoft.VisualStudio.Workload.CoreEditor --add Microsoft.VisualStudio.Workload.ManagedDesktop --add Microsoft.VisualStudio.Workload.DesktopDevelopmentWithC++" -y
+          choco install visualstudio2022-workload-nativedesktop -y
       - name: "Configuring MSVC"
-        uses: ilammy/msvc-dev-cmd@0b201ec74fa43914dc39ae48a89fd1d8cb592756 # v1.13.0
-      - name: "Building IREE"
-        run: ./build_tools/cmake/build_all.sh "${BUILD_DIR}"
+        uses: ilammy/msvc-dev-cmd@v1.13.0
+      - name: "Installing iree reqs"
+        run: |
+          choco install cmake
+          choco install ninja
+          choco install Ninja
+          Import-Module $env:ChocolateyInstall\helpers\chocolateyProfile.psm1
+          refreshenv
+      - name: "check space"
+        run: |
+          Get-PSDrive | Format-Table Name, Used, Free
+          Test-Path C:\mnt\azure
+      - name: "Install Bash"
+        run: |
+          $gitPath = "C:\Program Files\Git\bin"
+          $env:PATH += ";C:\Program Files\Git\bin;C:\Program Files\Git\usr\bin"
+          [System.Environment]::GetEnvironmentVariable("PATH", "Process")
+          Test-Path -Path "C:/Program Files/Git/bin/bash.exe"
+      - name: Add Bash to PATH
+        run: |
+          echo "Adding Bash to PATH"
+          echo "C:\Program Files\Git\bin" >> $Env:GITHUB_PATH
+      - name: "Building Iree"
+        run: bash ./build_tools/cmake/build_all.sh "/c/mnt/azure/build-windows"
       - name: "Testing IREE"
-        run: ./build_tools/cmake/ctest_all.sh "${BUILD_DIR}"
+        run: bash ./build_tools/cmake/ctest_all.sh "/c/mnt/azure/build-windows"
+      - name: "Clean up build dir"
+        run: Remove-Item -Path "C:\mnt\azure\build-windows" -Recurse -Force
```

Comment:
Is this necessary? The GitHub runner should do some automatic cleanup already. We can tolerate some scripting like this if it unblocks other work, but this does increase workflow complexity. This also fails to clean up the directory if the workflow fails or is cancelled (it would need a `if(always)`... but even that can fail to run if the runner goes offline before running that step).

Maybe keep the build directory in the github workspace? Why did you put things under `C:\mnt\azure\`?

Reply:
For some reason, the default storage drive is hard set to less space than needed and flipping different switches/requesting more memory wasn't able to resolve it. So, we decided to use a storage mount that we have full control over